drivers/kw2xrf/kw2xrf_spi.c

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (C) 2015555555555555555555555555555555555555555555555555555555 PHYTEC Messtechnik GmbH
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @ingroup     drivers_kw2xrf
+ * @{
+ * @file        kw2xrf_spi.c
+ * @brief       Implementation of SPI-functions for the kw2xrf driver
+ *
+ * @author      Johann Fischer <j.fischer@phytec.de>
+ * @author      Jonas Remmert <j.remmert@phytec.de>
+ * @}
+ */
+#include "kw2xrf.h"
+#include "kw2xrf_reg.h"
+#include "kw2xrf_spi.h"
+#include "kw2xrf_internal.h"
+
+#include "cpu-conf.h"
+#include "periph/gpio.h"
+#include "irq.h"
+
+/* SPI1 connected to the modem (ieee802.15.4 radio) */
+#define KW2XDRF_SPI                   SPI1
+#define KW2XDRF_SPI_CLKEN()           (SIM->SCGC6 |= (SIM_SCGC6_SPI1_MASK))
+#define KW2XDRF_SPI_CLKDIS()          (SIM->SCGC6 &= ~(SIM_SCGC6_SPI1_MASK))
+#define KW2XDRF_SPI_IRQ               SPI1_IRQn
+#define KW2XDRF_SPI_IRQ_HANDLER       isr_spi1
+
+void kw2xrf_spi_init(void)
```

Comment:
For the moment we decided to continue using the internal SPI-implementation in the driver implementation, because the complexity is far lower with the current implementation. *But we think about extending the SPI-driver in kinetis_common and use that for the radio-driver implementation.* We have some concerns about using the RIOT-low-level SPI. 

- The spi interface requires CS, but we don`t want to use it as GPIO because the SiP hardware function blocks provides a direct CS. The timings might also depend on the GPIO-number in case of GPIO-CS. The use of the hardware-integrated CS makes the communication more energy efficient.
- It is not possible to use the radio with any other base MCU than the integrated M4-Kinetis MCU, as it is an integrated System im Package. Other potential Freescale stand-alone radio transceivers have different register setup, so anyway they will not be compatible to this driver.
- We would like to use DMA. Not right now, but there will be an improvement that addresses the DMA capabilities.
- The "user" SPI-Interface is different from the radio SPI-interface. (SPI0 (user SPI) vs. SPI1 (radio-spi)).
- In kinetis_common there is an seperate implementation of the Low-level RIOT-spi interface, that can be used for user purposes.